src/api/ai_help.rs

```diff
@@ -596,3 +602,26 @@ pub async fn ai_help_delete_full_history(
     delete_full_help_history(&mut conn, &user)?;
     Ok(HttpResponse::Created().finish())
 }
+
+// This function is for QA purposes only, it triggering
+// an error based on the input message. The message can be optionally
+// set in the settings `ai.trigger_error_for_chat_term`. Nothing
+// will be triggered if the setting is missing, which should be the
+// situation in production-like environments.
+fn qa_check_for_error_trigger(
+    messages: &[ChatCompletionRequestMessage],
+) -> Result<(), crate::api::error::ApiError> {
+    if let Some(ref ai) = SETTINGS.ai {
+        if let Some(ref magic_words) = ai.trigger_error_for_chat_term {
+            let last_user_message = messages.iter().filter(|m| m.role == Role::User).last();
+            if let Some(msg) = last_user_message {
+                if let Some(ref msg_text) = msg.content {
+                    if msg_text == magic_words {
+                        return Err(crate::api::error::ApiError::Artificial);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
that's formatted so you can just accept if you think that's _more_ readable.
```suggestion
    if let Some(magic_words) = SETTINGS
        .ai
        .as_ref()
        .and_then(|ai| ai.trigger_error_for_chat_term.as_ref())
    {
        if let Some(msg_text) = messages
            .iter()
            .filter(|m| m.role == Role::User)
            .last()
            .and_then(|m| m.content.as_ref())
        {
            if msg_text == magic_words {
                return Err(crate::api::error::ApiError::Artificial);
            }
        }
    }
```